frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/common-template-validations/common-templates-validations.ts

```diff
@@ -45,15 +37,24 @@ export const getTemplateValidations = (
       ? iUserTemplates.find((template) => iGetName(template) === userTemplateName)
       : null;
 
+  if (iUserTemplate) {
+    const validations = getValidationsFromTemplates(List([iUserTemplate]));
+    return getFieldValidations(validations, IDToJsonPath[fieldId]);
+  }
+
+  // Get OS, workload-profile and flavor attributes
+  const os = iGetVmSettingAttribute(state, id, VMSettingsField.OPERATING_SYSTEM);
+  const flavor = iGetVmSettingAttribute(state, id, VMSettingsField.FLAVOR);
+  const workloadProfile = iGetVmSettingAttribute(state, id, VMSettingsField.WORKLOAD_PROFILE);
+
+  // Get all the templates from common-templates
+  const commonTemplates = iGetLoadedCommonData(state, id, VMWizardProps.commonTemplates);
+
   // Get all the validations from the relevant templates:
   // Get the validations from the user template, if chosen.
   // If not, get the validations from Common-Templates based on OS, Workload-Profile and Flavor
-  const validations = iUserTemplate
-    ? JSON.parse(iGetIn(iUserTemplate, ['metadata', 'annotations', 'validations']))
-    : getValidationsFromTemplates(
-        // Get templates based on OS, Workload-Profile and Flavor
-        getRelevantTemplates(commonTemplates, os, workloadProfile, flavor),
-      );
+  const templates = getRelevantTemplates(commonTemplates, os, workloadProfile, flavor);
+  const validations = getValidationsFromTemplates(templates);
 
   // Return all the validations which are relevant for the field
   return getFieldValidations(validations, IDToJsonPath[fieldId]);
```

Comment:
also at https://github.com/openshift/console/pull/3879/files#diff-c3d6503935eeb5d51e6cca027270cb72R75
- is it possible for a template to have two standalone validations? One min and the other max?
- shouldn't we have a default validation.message if it is missing one?
- shouldn't we rather use custom messages where we can. E.g. memory validation `This VM requires more memory.` is not much fitting for the wizard IMO. We should at least say what the minimal ammount is.

Reply:
> is it possible for a template to have two standalone validations? One min and the other max?

I is grammatically possible, but is semantically nonsense, for example, one `min` and one `max` rule, will always pass validation whatever the values.

> shouldn't we have a default validation.message if it is missing one?

Nice, fixing

> shouldn't we rather use custom messages where we can. E.g. memory validation This VM requires more memory. is not much fitting for the wizard IMO. We should at least say what the minimal ammount is.

Makes sense to me too ... fixing.

Reply:
> > is it possible for a template to have two standalone validations? One min and the other max?
> 
> I is grammatically possible, but is semantically nonsense, for example, one `min` and one `max` rule, will always pass validation whatever the values.
> 

why nonsense? Min 1Gi and max 4Gi  should check for both sides of the interval, no?

Reply:
> why nonsense? Min 1Gi and max 4Gi should check for both sides of the interval, no?

In this case, any value is either above 1G **or** below 4G, so it's always true.
if users wants to check for between 1G and 4G they need to use the min/max rule.

Reply:
why is it **or**? Can you point me to the specification?
Wouldn't it make more sense to combine the validations with **and**? 

I agree min/max in one validation is better, but should we expect that the users will always use it this way?

Reply:
> why is it or? 

The reasoning is:

We start from a none specific state where we match several templates ( for example all "windows" templates) in this state, all memory values matching at least one template is valid.

As we get more specific ( for example windows server ) we have less templates, but again, if at least one template is possible, the value is valid.

And when we have only one template, we will probable have only one rule ... ( min, max or min/max ), any combination of this rules is a miss configuration on the side of the template writer ( p.s. all the memory rules in common templates are `min`, and it makes sense too ... ) 

> Can you point me to the specification?

multiple templates is outside the scope of the docs, IMHO we need to do what makes more sense.

> Wouldn't it make more sense to combine the validations with and?

no, for example if `windows-server` need `min=4G` and `windows-desktop` need `min=2G`, in the state where we still don't know what template we end up with, we should allow the server **or** desktop, and recheck when we know what workload was chosen.

> I agree min/max in one validation is better, but should we expect that the users will always use it this way?

yes, 
a- min, max, and min/max [have a defined meaning](https://github.com/kubevirt/common-templates/blob/master/templates/VALIDATION.md#min--max) as one rule are defined, having two rules of min and max for same path is not defined.
b- all common template use one rule of min,max,min/max per path ( actually they use only min)

Reply:
> > why is it or?
> 
> The reasoning is:
> 
> We start from a none specific state where we match several templates ( for example all "windows" templates) in this state, all memory values matching at least one template is valid.
> 
> As we get more specific ( for example windows server ) we have less templates, but again, if at least one template is possible, the value is valid.
> 

yup, true

> And when we have only one template, we will probable have only one rule ... ( min, max or min/max ), any combination of this rules is a miss configuration on the side of the template writer ( p.s. all the memory rules in common templates are `min`, and it makes sense too ... )

> 
> > Can you point me to the specification?
> 
> multiple templates is outside the scope of the docs, IMHO we need to do what makes more sense.

I was talking about the `miss configuration on the side of the template writer` from the beginning. Do we have a specification for that?

One can write a user template with these two rules and think it is correct.

> 
> > Wouldn't it make more sense to combine the validations with and?
> 
> no, for example if `windows-server` need `min=4G` and `windows-desktop` need `min=2G`, in the state where we still don't know what template we end up with, we should allow the server **or** desktop, and recheck when we know what workload was chosen.

not what I was talking about - just one single template with multiple validations

> 
> > I agree min/max in one validation is better, but should we expect that the users will always use it this way?
> 
> yes,
> a- min, max, and min/max [have a defined meaning](https://github.com/kubevirt/common-templates/blob/master/templates/VALIDATION.md#min--max) as one rule are defined, having two rules of min and max for same path is not defined.

I think we should support that if the behavior is not defined. 

IMO it is more sane to behave like that. It is an array of validations so one would expect that all of them will execute.


> b- all common template use one rule of min,max,min/max per path ( actually they use only min)

I know, this concerns just user templates now